compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1826,6 +1832,258 @@ public void visit(BLangWorkerReceive workerReceiveExpr) {
         resultType = this.expType;
     }
 
+    @Override
+    public void visit(BLangWorkerMultipleReceive workerMultipleReceive) {
+        switch (expType.tag) {
+            case TypeTags.RECORD:
+                checkTypesForWorkerMultipleReceiveRecords(workerMultipleReceive);
+                break;
+            case TypeTags.MAP:
+                if (checkTypeForWorkerMultipleMap(workerMultipleReceive.receiveFields)) {
+
+                    resultType = this.expType;
+                }
+                break;
+        }
+    }
+
+    private void checkTypesForWorkerMultipleReceiveRecords(BLangWorkerMultipleReceive bLangWorkerMultipleReceive) {
+
+        Map<String, BField> lhsFields = ((BRecordType) expType).fields;
+
+        // check if the record is sealed, if so check if the fields in wait collection is more than the fields expected
+        // by the lhs record
+        if (((BRecordType) expType).sealed && bLangWorkerMultipleReceive.receiveFields.size() > lhsFields.size()) {
+            dlog.error(bLangWorkerMultipleReceive.pos, DiagnosticCode.INCOMPATIBLE_TYPES, expType,
+                    getWorkerMultipleReceiveReturnType(bLangWorkerMultipleReceive.receiveFields));
+            resultType = symTable.semanticError;
+            return;
+
+        }
+
+        for (WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField : bLangWorkerMultipleReceive.receiveFields) {
+
+            String key;
+            key = receiveField.getWorkerFieldName() != null ?
+                    receiveField.getWorkerFieldName().getValue() : receiveField.getWorkerName().getValue();
+            if (!lhsFields.containsKey(key)) {
+                // Check if the field is sealed, if so you cannot have dynamic fields
+                if (((BRecordType) expType).sealed) {
+                    dlog.error(bLangWorkerMultipleReceive.pos, DiagnosticCode.INVALID_FIELD_NAME_RECORD_LITERAL, key,
+                            expType);
+                    resultType = symTable.semanticError;
```

Comment:
Can we return here and remove `if (symTable.semanticError != resultType)` in L1889?

Reply:
If we return here, it wouldn't complete the for-loop and execute the `checkMissingReqFieldsForWait(((BRecordType) expType), rhsFields, waitExpr.pos);`